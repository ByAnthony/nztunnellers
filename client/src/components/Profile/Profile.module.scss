@use 'src/index.scss';

.container {
  @include index.component-margin;
}

.profile {
  @include index.component-margin;

  @include index.desktop {
    display: flex;
    margin: 0 auto;
    max-width: index.$inner-width;
    gap: index.$l-margin;
  }

  .flex-summary {
    @include index.desktop {
      width: 40%;
    }
  }

  .summary {
    @include index.desktop {
      flex-grow: 1;
      text-align: left;
    }
  }

  .flex-diary {
    @include index.desktop {
      width: 60%;
    }
  }
}
