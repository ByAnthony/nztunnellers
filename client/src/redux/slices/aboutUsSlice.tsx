import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { AboutUs } from '../../types/about-us';

export const aboutUsApi = createApi({
  reducerPath: 'aboutUsApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'http://localhost:5000/',
  }),
  endpoints: (builder) => ({
    getAboutUs: builder.query<AboutUs, void>({
      query: () => ({
        url: '/about-us',
      }),
    }),
  }),
});

export const {
  useGetAboutUsQuery,
} = aboutUsApi;
