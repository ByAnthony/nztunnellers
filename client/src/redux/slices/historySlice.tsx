import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { Article, Articles } from '../../types/article';

export const historyApi = createApi({
  reducerPath: 'historyApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'http://localhost:5000/',
  }),
  endpoints: (builder) => ({
    getAllHistoryArticleLink: builder.query<Array<Articles>, void>({
      query: () => ({
        url: '/',
      }),
    }),
    getHistoryArticleById: builder.query<Article, string>({
      query: (id: string) => `history/${id}`,
    }),
  }),
});

export const {
  useGetAllHistoryArticleLinkQuery,
  useGetHistoryArticleByIdQuery,
} = historyApi;
