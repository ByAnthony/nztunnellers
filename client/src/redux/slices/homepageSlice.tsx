import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { Articles } from '../../types/article';

export const homepageApi = createApi({
  reducerPath: 'homepageApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'http://localhost:5000/',
  }),
  endpoints: (builder) => ({
    getHomepageData: builder.query<Record<string, Array<Articles>>, void>({
      query: () => ({
        url: '/',
      }),
    }),
  }),
});

export const {
  useGetHomepageDataQuery,
} = homepageApi;
