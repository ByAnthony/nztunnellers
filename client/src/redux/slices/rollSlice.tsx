import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { Details } from '../../types/roll';
import { Profile } from '../../types/tunneller';

export const tunnellersApi = createApi({
  reducerPath: 'tunnellersApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'http://localhost:5000/',
  }),
  endpoints: (builder) => ({
    getAllTunnellers: builder.query<Record<string, Array<Details>>, void>({
      query: () => ({
        url: 'roll/',
      }),
    }),
    getTunnellerById: builder.query<Profile, number>({
      query: (id: number) => `roll/${id}`,
    }),
  }),
});

export const { useGetAllTunnellersQuery, useGetTunnellerByIdQuery } = tunnellersApi;
