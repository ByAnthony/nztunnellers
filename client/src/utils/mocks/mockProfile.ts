// import { Profile } from '../../types/tunneller';
// import { mockSummary } from './mockSummary';

export const mockId = 26;

// export const mockProfile = (overrides: Partial<Profile> | undefined = undefined): Profile => ({
//   id: mockId,
//   summary: mockSummary(),
//   origins: mockOrigins(),
//   preWarYears: mockPreWarYears(),
//   militaryYears: mockMilitaryYears(),
//   postServiceYears: mockPostServiceYears(),
//   sources: mockSources(),
//   image: mockImage(),
//   ...overrides,
// });
