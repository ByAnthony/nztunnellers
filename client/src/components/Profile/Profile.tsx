import { useParams } from 'react-router-dom';
import { useGetTunnellerByIdQuery } from '../../redux/slices/rollSlice';
import { Menu } from '../Menu/Menu';
import { ProfileHowToCite } from '../ProfileHowToCite/ProfileHowToCite';
import { ProfileOverview } from '../ProfileOverview/ProfileOverview';
import { ProfileDiary } from '../ProfileDiary/ProfileDiary';
import { ProfileSources } from '../ProfileSources/ProfileSources';
import { ProfileSummary } from '../ProfileSummary/ProfileSummary';
import STYLES from './Profile.module.scss';

export function Profile() {
  const { id } = useParams();
  const {
    data, error, isLoading, isSuccess,
  } = useGetTunnellerByIdQuery(Number(id!));

  if (data) {
    return (
      <>
        {error && (
        <div className={STYLES['profile-container']}>
          <p>An error occured</p>
        </div>
        )}
        <Menu />
        { isLoading }
        { isSuccess && (
        <div className={STYLES['profile-container']}>
          <div className={STYLES['sticky-card']}>
            <div className={STYLES.summary}>
              <ProfileSummary
                summary={data.summary}
                embarkationUnit={data.militaryYears.embarkationUnit}
                enlistment={data.militaryYears.enlistment}
              />
            </div>
          </div>
          <div>
            <ProfileOverview
              id={data.id}
              name={data.summary.name}
              militaryYears={data.militaryYears}
            />
            <ProfileDiary
              tunnellerId={data.id}
              origins={data.origins}
              preWarYears={data.preWarYears}
              militaryYears={data.militaryYears}
              postWarYears={data.postServiceYears}
            />
            <ProfileSources sources={data.sources} />
            <ProfileHowToCite id={Number(id)} summary={data.summary} />
          </div>
        </div>
        )}
      </>
    );
  }
  return null;
}
