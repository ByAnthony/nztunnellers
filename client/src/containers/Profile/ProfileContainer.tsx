import { useGetTunnellerByIdQuery } from "../../redux/slices/rollSlice";
import { useParams } from "react-router-dom";

export const ProfileContainer = () => {

  const { id } = useParams();
  const tunneller = useGetTunnellerByIdQuery(Number(id!));

  return (
    <div>
     {tunneller.data?.name === undefined ? 'not found' : tunneller.data.name.forename + tunneller.data.name.surname}
    </div>
  );
}
