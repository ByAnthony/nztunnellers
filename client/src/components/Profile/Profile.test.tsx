import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { mockProfile } from '../../utils/mocks/mockProfile';
import { useGetTunnellerByIdQuery } from '../../redux/slices/rollSlice';
import { Profile } from './Profile';

jest.mock('../../redux/slices/rollSlice', () => ({
  useGetTunnellerByIdQuery: jest.fn(),
}));

jest.mock('../../utils/date-utils', () => ({
  today: new Date('2023-05-04'),
}));

describe('Profile', () => {
  it('renders profile when data is available', () => {
    (useGetTunnellerByIdQuery as jest.Mock).mockReturnValue({
      data: mockProfile,
      error: null,
      isLoading: false,
      isSuccess: true,
    });

    const { asFragment } = render(<Profile />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('does not render profile when data is undefined', () => {
    (useGetTunnellerByIdQuery as jest.Mock).mockReturnValue({
      data: undefined,
      error: null,
      isLoading: false,
      isSuccess: true,
    });

    const { container } = render(<Profile />);

    expect(container).toBeEmptyDOMElement();
  });
});
