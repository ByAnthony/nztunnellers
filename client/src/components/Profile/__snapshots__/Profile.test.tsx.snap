// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders profile when data is available 1`] = `
<DocumentFragment>
  <div
    class="menu"
  >
    <a
      aria-label="Go to the Homepage."
      class="logo"
      href="/"
    >
      <img
        alt=""
        src="/nzt_logo.png"
      />
    </a>
    <div
      class="search-form"
    >
      <form>
        <input
          id="search"
          name="search"
          placeholder="Search for a tunneller"
          type="text"
        />
      </form>
      <button
        class="search-form-button"
        type="submit"
      >
        <img
          alt=""
          height="18"
          src="/searching_hover.png"
        />
      </button>
    </div>
  </div>
  <div
    class="header"
  >
    <div
      class="link"
    >
      <a
        href="/tunnellers"
      >
        Tunnellers
      </a>
      <span>
        /
      </span>
      John Doe
    </div>
    <h1>
      <span
        class="forename"
      >
        John
      </span>
      <span
        class="surname"
      >
        Doe
      </span>
    </h1>
    <p
      class="dates"
    >
      1888 - 1975
    </p>
  </div>
  <div
    class="profile"
  >
    <div
      class="flex-summary"
    >
      <div
        class="summary"
      >
        <div
          class="overview"
        >
          <div
            class="image-card"
          >
            <img
              alt="Portrait of John Doe"
              src="/images/roll/tunnellers/1/1234.jpg"
            />
          </div>
          <div
            class="fullwidth-main-card"
          >
            <p>
              Unit
            </p>
            <span>
              Main Body
            </span>
          </div>
          <div
            class="fullwidth-main-card"
          >
            <p>
              Section
            </p>
            <span>
              Section No.2
            </span>
          </div>
          <div
            class="halfwidth-cards-container"
          >
            <div
              class="halfwidth-secondary-card"
            >
              <div
                class="halfwidth-secondary-card-title"
              >
                <p>
                  Rank
                </p>
              </div>
              <div>
                <span>
                  Sapper
                </span>
              </div>
            </div>
            <div
              class="halfwidth-secondary-card"
            >
              <div
                class="halfwidth-secondary-card-title"
              >
                <p>
                  Serial
                </p>
              </div>
              <div>
                <span>
                  1/1000
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex-diary"
    >
      <div
        class="diary"
      >
        <h2>
          About
        </h2>
        <div
          class="fullwidth-main-card"
        >
          <p>
            Born in New Zealand
          </p>
          <span>
            18 December 1886
          </span>
        </div>
        <div
          class="fullwidth-main-card"
        >
          Parents
        </div>
        <div
          class="halfwidth-cards-container"
        >
          <div
            class="halfwidth-secondary-card"
          >
            <div
              class="halfwidth-secondary-card-title"
            >
              <p>
                Mother
              </p>
            </div>
            <div>
              <span>
                Jane Doe
              </span>
            </div>
          </div>
          <div
            class="halfwidth-secondary-card"
          >
            <div
              class="halfwidth-secondary-card-title"
            >
              <p>
                Father
              </p>
            </div>
            <div>
              <span>
                John Doe
              </span>
            </div>
          </div>
        </div>
        <div
          class="fullwidth-main-card"
        >
          Work
        </div>
        <div
          class="halfwidth-cards-container"
        >
          <div
            class="halfwidth-secondary-card"
          >
            <div
              class="halfwidth-secondary-card-title"
            >
              <p>
                Occupation
              </p>
            </div>
            <div>
              <span>
                Goldminer
              </span>
            </div>
          </div>
          <div
            class="halfwidth-secondary-card"
          >
            <div
              class="halfwidth-secondary-card-title"
            >
              <p>
                Employer
              </p>
            </div>
            <div>
              <span>
                Goldmining Company
              </span>
            </div>
          </div>
        </div>
        <h2>
          Army Experience
        </h2>
        <li
          class="fullwidth-secondary-card"
        >
          <span>
            NZ Infantry
          </span>
          <p>
            12 months in New Zealand
          </p>
        </li>
        <a
          aria-label="Open the World War I timeline."
          class="war-service"
          href="/tunnellers/26/wwi-timeline"
        >
          <div>
            <p>
              World War I (1914-1918)
            </p>
            <span>
              New Zealand Tunnellers
            </span>
          </div>
          <div
            class="arrow-right"
          >
            →
          </div>
        </a>
        <div
          class="halfwidth-cards-container"
        >
          <div
            class="halfwidth-secondary-card"
          >
            <div
              class="halfwidth-secondary-card-title"
            >
              <p>
                <img
                  alt="British War Medal ribbon"
                  src="/images/roll/medals/british-war-medal.png"
                  width="40"
                />
              </p>
            </div>
            <div>
              <span>
                British War Medal
              </span>
            </div>
          </div>
          <div
            class="halfwidth-secondary-card"
          >
            <div
              class="halfwidth-secondary-card-title"
            >
              <p>
                <img
                  alt="Victory Medal ribbon"
                  src="/images/roll/medals/victory-medal.png"
                  width="40"
                />
              </p>
            </div>
            <div>
              <span>
                Victory Medal
              </span>
            </div>
          </div>
        </div>
        <h2>
          Death
        </h2>
        <div
          class="fullwidth-main-card"
        >
          <p>
            Died at the age of 89
          </p>
          <span>
            13 October 1926
          </span>
        </div>
      </div>
      <div
        class="sources"
      >
        <h3>
          Sources
        </h3>
        <p>
          Auckland War Memorial Museum Tāmaki Paenga Hira: 
          <a
            href="https://www.mockurl.co.nz/online-cenotaph/B2874930"
          >
            Online Cenotaph He Toa Taumata Rau
          </a>
          .
        </p>
        <p>
          New Zealand Archives Te Rua Mahara o te Kāwanatanga, AABK 18805 W5530 39/0022386, 
          <a
            href="https://www.mockurl.co.nz/nzarchives/B2874930"
          >
            Military Personnel File
          </a>
          .
        </p>
        <p>
          <em>
            London Gazette, 
          </em>
          23 May 1917, p. 1675.
        </p>
        <p>
          <em>
            Nominal Roll of New Zealand Expeditionary Force, 1915. New Zealand Engineers Tunnelling Company
          </em>
          , Government Printer, Wellington, 1916, 37.
        </p>
      </div>
      <div
        class="howtocite"
      >
        <h3>
          How to cite this page
        </h3>
        <p>
          Anthony Byledbal, "John Doe (1888-1975)", 
          <em>
            New Zealand Tunnellers Website
          </em>
          , 2023 (2009), Accessed: 4 May 2023. URL: www.nztunnellers.com/tunnellers/26.
        </p>
      </div>
    </div>
  </div>
  <div
    class="footer"
  >
    <div
      class="map"
    >
      <div
        class="links"
      >
        <div
          class="link"
        >
          <a
            href="/history"
          >
            History
          </a>
        </div>
        <div
          class="link"
        >
          <a
            href="/tunnellers"
          >
            Tunnellers
          </a>
        </div>
        <div
          class="link"
        >
          <a
            href="/about-us"
          >
            About Us
          </a>
        </div>
      </div>
      <button
        aria-label="Go back to the top of the page."
        class="scroll-top"
        type="button"
      >
        ↑
      </button>
    </div>
    <div
      class="support"
    >
      <div>
        <a
          aria-label="Go to The Artois University website."
          href="https://www.univ-artois.fr/artois-university"
        >
          <img
            alt="Artois University homepage"
            src="/images/support/logo-univ-artois-blanc_0.png"
          />
        </a>
      </div>
      <div>
        <a
          aria-label="Go to The Institute for Strategic Research website."
          href="https://www.irsem.fr/en/"
        >
          <img
            alt="Institute for Strategic Research homepage"
            src="/images/support/irsem-white.png"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
